test/Sentry.AspNetCore.Tests/SentryMiddlewareTests.cs

```diff
@@ -479,5 +479,37 @@ public void PopulateScope_Sdk_ContainNameAndVersion()
             Assert.Equal(Constants.SdkName, scope.Sdk.Name);
             Assert.Equal(SentryMiddleware.NameAndVersion.Version, scope.Sdk.Version);
         }
+
+        [Fact]
+        public async Task InvokeAsync_DefaultOptions_CanNotFlush()
+        {
+            var sut = _fixture.GetSut();
+            await sut.InvokeAsync(_fixture.HttpContext);
+
+            await _fixture.Hub.FlushAsync(Arg.Any<TimeSpan>()).DidNotReceive();
+        }
+
+        [Fact]
+        public async Task InvokeAsync_FlushOnCompletedRequestWhenFalse_CanNotFlush()
+        {
+            var sut = _fixture.GetSut(); 
+            _fixture.Options.FlushOnCompletedRequest = false;
+
+            await sut.InvokeAsync(_fixture.HttpContext);
+
+            await _fixture.Hub.FlushAsync(Arg.Any<TimeSpan>()).DidNotReceive();
+        }
+
+        [Fact]
+        public async Task InvokeAsyn_FlushOnCompletedRequestTrue_RespectsTimeout()
+        {
+            var sut = _fixture.GetSut();
+            _fixture.Options.FlushOnCompletedRequest = true;
+            _fixture.Options.FlushTimeout = TimeSpan.FromSeconds(1);
+
+            await sut.InvokeAsync(_fixture.HttpContext);
+
+            await _fixture.Hub.FlushAsync(Arg.Any<TimeSpan>()).Received();
```

Comment:
With `Arg.Any` you match with any timeout being passed in.

To make sure the configuration you've defined above, try to assert with the exact arguments:

```suggestion
            await _fixture.Hub.FlushAsync(_fixture.Options.FlushTimeout).Received(1);
```

Reply:
Oh the order here is inverted I believe:
```suggestion
            await _fixture.Hub.Received().FlushAsync(Arg.Any<TimeSpan>());
```